packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -12,18 +12,48 @@ import CustomSelectItem from '../item';
 import type { LegacyCustomSelectProps } from '../types';
 import * as Styled from '../styles';
 
+function useDeprecatedProps( {
+	showSelectedHint = false,
+	options,
+	// Deprecated
+	__experimentalShowSelectedHint,
+	...otherProps
+}: LegacyCustomSelectProps ) {
+	return {
+		...otherProps,
+		options: options.map( ( o ) => {
+			const toReturn = o;
+
+			const hint =
+				typeof o.__experimentalHint !== 'undefined'
+					? o.__experimentalHint
+					: o.hint;
+
+			if ( hint ) {
+				toReturn.hint = hint;
+			}
+
+			return toReturn;
+		} ),
+		showSelectedHint:
+			typeof __experimentalShowSelectedHint !== 'undefined'
+				? __experimentalShowSelectedHint
+				: showSelectedHint,
+	};
```

Comment:
This prioritizes the `__experimental` over the stable prop. Would it be better to do it like this, so the stable prop is prioritized?

```js
function useDeprecatedProps( {
  __experimentalShowSelectedHint,
  ...otherProps
} ) {
  return {
    showSelectedHint: __experimentalShowSelectedHint,
    ...otherProps
  }
}
```

Reply:
Good point, done!